src/Mvc/Mvc.Analyzers/src/Startup/StartupAnalyzer.Diagnostics.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    public partial class StartupAnalzyer : DiagnosticAnalyzer
+    {
+        internal readonly static DiagnosticDescriptor UnsupportedUseMvcWithEndpointRouting = new DiagnosticDescriptor(
+            "MVC1005", 
+            "Cannot use UseMvc with Endpoint Routing.", 
+            "Using '{0}' to configure MVC is not supported while using Endpoint Routing. To continue using '{0}', please set 'MvcOptions.EnableEndpointRounting = false' inside '{1}'.",
+            "Usage", 
+            DiagnosticSeverity.Warning, 
```

Comment:
I could go either way. I lean towards warning because we're going to throw anyway.

Reply:
I think it should be a warning at most. This will never be 100% perfect. A dev might have set the flag in a way that we don't detect.

A warning generally won't block someone, and if they take some time to clean up any warnings they have they can suppress it then.

Reply:
The answer won't ever be incredibly cut and dry btw. You can configure analyzers so, you can actually change the level of any error, in addition to suppressing the warning.

Reply:
I'm not super sure it this applies to user created analyzers, but errors produced by Roslyn's stylistic code fixes (such as use throws expression instead of if checks) show up as errors in the error window, but don't actually fail the build.

Reply:
The stylistic ones are **definitely** not errors, they are `Info`. https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

Reply:
My opinon btw is the warning is the best value for what we intend.

Reply:
+1